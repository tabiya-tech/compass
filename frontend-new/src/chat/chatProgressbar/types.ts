export enum ConversationPhase {
  INITIALIZING = "INITIALIZING",
  INTRO = "INTRO",
  COLLECT_EXPERIENCES = "COLLECT_EXPERIENCES",
  DIVE_IN = "DIVE_IN",
  ENDED = "ENDED",
  UNKNOWN = "UNKNOWN"
}

export type CurrentPhase = {
  percentage: number
  phase: ConversationPhase
}

export const defaultCurrentPhase = {
  phase: ConversationPhase.INITIALIZING,
  percentage: 0
}
