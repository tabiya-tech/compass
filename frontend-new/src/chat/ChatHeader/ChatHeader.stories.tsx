import type { Meta, StoryObj } from "@storybook/react";
import ChatHeader from "./ChatHeader";

const meta: Meta<typeof ChatHeader> = {
  title: "Chat/ChatHeader",
  component: ChatHeader,
  tags: ["autodocs"],
  argTypes: {
    notifyOnLogout: { action: "notifyOnLogout" },
    notifyOnExperiencesDrawerOpen: { action: "notifyOnExperiencesDrawerOpen" },
  },
};

export default meta;

type Story = StoryObj<typeof ChatHeader>;

export const Shown: Story = {
  args: {},
};
