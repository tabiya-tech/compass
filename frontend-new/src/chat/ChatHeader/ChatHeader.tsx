import React, { useContext, useState } from "react";
import { Box, Typography, useTheme } from "@mui/material";
import { NavLink, useNavigate } from "react-router-dom";
import { routerPaths } from "src/app/routerPaths";
import { MenuItemConfig } from "src/theme/ContextMenu/menuItemConfig.types";
import PermIdentityIcon from "@mui/icons-material/PermIdentity";
import PrimaryIconButton from "src/theme/PrimaryIconButton/PrimaryIconButton";
import ContextMenu from "src/theme/ContextMenu/ContextMenu";
import { IsOnlineContext } from "src/app/isOnlineProvider/IsOnlineProvider";
import * as Sentry from "@sentry/react";

export type ChatHeaderProps = {
  notifyOnLogout: () => void;
  startNewConversation: () => void;
  notifyOnExperiencesDrawerOpen: () => void;
};

const uniqueId = "7413b63a-887b-4f41-b930-89e9770db12b";
export const DATA_TEST_ID = {
  CHAT_HEADER_CONTAINER: `chat-header-container-${uniqueId}`,
  CHAT_HEADER_LOGO: `chat-header-logo-${uniqueId}`,
  CHAT_HEADER_LOGO_LINK: `chat-header-logo-link-${uniqueId}`,
  CHAT_HEADER_ICON_USER: `chat-header-icon-user-${uniqueId}`,
  CHAT_HEADER_BUTTON_USER: `chat-header-button-user-${uniqueId}`,
};

export const MENU_ITEM_ID = {
  SETTINGS_SELECTOR: `settings-selector-${uniqueId}`,
  LOGOUT_BUTTON: `logout-button-${uniqueId}`,
  START_NEW_CONVERSATION: `start-new-conversation-${uniqueId}`,
  EXPERIENCES_BUTTON: `experiences-button-${uniqueId}`,
  REPORT_BUG_BUTTON: `report-bug-button-${uniqueId}`,
};

export const MENU_ITEM_TEXT = {
  SETTINGS: "settings",
  LOGOUT: "logout",
  START_NEW_CONVERSATION: "start new conversation",
  EXPERIENCES: "view experiences",
  REPORT_BUG: "report a bug",
};

const ChatHeader: React.FC<Readonly<ChatHeaderProps>> = ({
  notifyOnLogout,
  startNewConversation,
  notifyOnExperiencesDrawerOpen,
}) => {
  const theme = useTheme();
  const navigate = useNavigate();
  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);

  const isOnline = useContext(IsOnlineContext);
  const contextMenuItems: MenuItemConfig[] = [
    {
      id: MENU_ITEM_ID.START_NEW_CONVERSATION,
      text: MENU_ITEM_TEXT.START_NEW_CONVERSATION,
      disabled: !isOnline,
      action: startNewConversation,
    },
    {
      id: MENU_ITEM_ID.EXPERIENCES_BUTTON,
      text: MENU_ITEM_TEXT.EXPERIENCES,
      disabled: false,
      action: notifyOnExperiencesDrawerOpen,
    },
    {
      id: MENU_ITEM_ID.SETTINGS_SELECTOR,
      text: MENU_ITEM_TEXT.SETTINGS,
      disabled: !isOnline,
      action: () => {
        navigate(routerPaths.SETTINGS);
      },
    },
    {
      id: MENU_ITEM_ID.REPORT_BUG_BUTTON,
      text: MENU_ITEM_TEXT.REPORT_BUG,
      disabled: !isOnline,
      action: () => {
        const feedback = Sentry.getFeedback();
        if (feedback) {
          feedback.createForm().then((form) => {
            if (form) {
              form.appendToDom();
              form.open();
            }
          });
        }
      },
    },
    {
      id: MENU_ITEM_ID.LOGOUT_BUTTON,
      text: MENU_ITEM_TEXT.LOGOUT,
      disabled: !isOnline,
      action: notifyOnLogout,
    },
  ];

  return (
    <Box
      display="flex"
      alignItems="center"
      justifyContent="space-between"
      data-testid={DATA_TEST_ID.CHAT_HEADER_CONTAINER}
      padding={theme.spacing(theme.tabiyaSpacing.md)}
    >
      <NavLink style={{ lineHeight: 0 }} to={routerPaths.ROOT} data-testid={DATA_TEST_ID.CHAT_HEADER_LOGO_LINK}>
        <img
          src={`${process.env.PUBLIC_URL}/compass.svg`}
          alt="Compass"
          height="48px"
          data-testid={DATA_TEST_ID.CHAT_HEADER_LOGO}
        />
      </NavLink>
      <Typography variant="h1">Compass</Typography>
      <Box
        sx={{
          display: "flex",
          alignItems: "center",
          flexDirection: "row",
          gap: theme.spacing(theme.tabiyaSpacing.md),
        }}
      >
        <PrimaryIconButton
          sx={{
            color: theme.palette.common.black,
          }}
          onClick={(event) => setAnchorEl(event.currentTarget)}
          data-testid={DATA_TEST_ID.CHAT_HEADER_BUTTON_USER}
          title="user info"
        >
          <PermIdentityIcon data-testid={DATA_TEST_ID.CHAT_HEADER_ICON_USER} />
        </PrimaryIconButton>
      </Box>
      <ContextMenu
        anchorEl={anchorEl}
        open={Boolean(anchorEl)}
        notifyOnClose={() => setAnchorEl(null)}
        items={contextMenuItems}
      />
    </Box>
  );
};

export default ChatHeader;
