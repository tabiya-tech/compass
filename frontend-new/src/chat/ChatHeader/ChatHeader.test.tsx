// mute the console
import "src/_test_utilities/consoleMock";

import ChatHeader, { DATA_TEST_ID, MENU_ITEM_ID } from "./ChatHeader";
import { render, screen } from "src/_test_utilities/test-utils";
import { act, fireEvent, waitFor, within } from "@testing-library/react";
import { HashRouter, useNavigate } from "react-router-dom";
import { routerPaths } from "src/app/routerPaths";
import { testNavigateToPath } from "src/_test_utilities/routeNavigation";
import ContextMenu from "src/theme/ContextMenu/ContextMenu";
import { MenuItemConfig } from "src/theme/ContextMenu/menuItemConfig.types";
import { mockBrowserIsOnLine } from "src/_test_utilities/mockBrowserIsOnline";

// mock the ContextMenu
jest.mock("src/theme/ContextMenu/ContextMenu", () => {
  return {
    __esModule: true,
    default: jest.fn(({ items }: { items: MenuItemConfig[] }) => (
      <div data-testid="mock-context-menu">
        {items.map((item) => (
          <div key={item.id} data-testid={item.id} onClick={item.action}>
            {item.text}
          </div>
        ))}
        ;
      </div>
    )),
  };
});

// mock the router
jest.mock("react-router-dom", () => {
  const actual = jest.requireActual("react-router-dom");
  return {
    ...actual,
    __esModule: true,
    useNavigate: jest.fn().mockReturnValue(jest.fn()),
  };
});

// mock the SocialAuthServices
jest.mock("src/auth/services/socialAuth/SocialAuth.service", () => {
  return {
    __esModule: true,
    default: jest.fn().mockImplementation(() => {
      return {
        logout: jest.fn(),
      };
    }),
  };
});

describe("ChatHeader", () => {
  beforeEach(() => {
    (ContextMenu as jest.Mock).mockClear();
  });

  test("should render the Chat Header", () => {
    // GIVEN a ChatHeader component
    const givenNotifyOnLogout = jest.fn();
    const givenStartNewConversation = jest.fn();
    const givenNotifyOnExperiencesDrawerOpen = jest.fn();
    const givenChatHeader = (
      <HashRouter>
        <ChatHeader
          notifyOnLogout={givenNotifyOnLogout}
          startNewConversation={givenStartNewConversation}
          notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
        />
      </HashRouter>
    );

    // WHEN the chat header is rendered
    render(givenChatHeader);

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the chat header to be visible
    expect(screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_CONTAINER)).toBeInTheDocument();
    // AND the chat header logo to be visible
    expect(screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_LOGO)).toBeInTheDocument();
    // AND the Compass text to be visible
    expect(screen.getByText("Compass")).toBeInTheDocument();
    // AND the user button to be shown with the user icon
    const chatHeaderButton = screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_BUTTON_USER);
    expect(chatHeaderButton).toBeInTheDocument();
    const chatHeaderUserIcon = within(chatHeaderButton).getByTestId(DATA_TEST_ID.CHAT_HEADER_ICON_USER);
    expect(chatHeaderUserIcon).toBeInTheDocument();
    // AND to match the snapshot
    expect(screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_CONTAINER)).toMatchSnapshot();
  });
  describe("chatHeader action tests", () => {
    const givenNotifyOnLogout = jest.fn();
    const givenStartNewConversation = jest.fn();
    const givenNotifyOnExperiencesDrawerOpen = jest.fn();
    const givenChatHeader = (
      <ChatHeader
        notifyOnLogout={givenNotifyOnLogout}
        startNewConversation={givenStartNewConversation}
        notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
      />
    );
    testNavigateToPath(givenChatHeader, "Compass logo", DATA_TEST_ID.CHAT_HEADER_LOGO_LINK, routerPaths.ROOT);

    test("should open the context menu when the user icon is clicked", async () => {
      // GIVEN a ChatHeader component
      const givenNotifyOnLogout = jest.fn();
      const givenChatHeader = (
        <HashRouter>
          <ChatHeader
            notifyOnLogout={givenNotifyOnLogout}
            startNewConversation={givenStartNewConversation}
            notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
          />
        </HashRouter>
      );
      // AND the chat header is rendered
      render(givenChatHeader);

      // WHEN the user button is clicked
      const userButton = screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_BUTTON_USER);
      fireEvent.click(userButton);

      // THEN expect the context menu to be visible
      await waitFor(() => {
        expect(ContextMenu).toHaveBeenCalledWith(
          expect.objectContaining({
            anchorEl: userButton,
            open: true,
          }),
          {}
        );
      });
    });

    test("should navigate to settings when the settings menu item is clicked", async () => {
      // GIVEN a ChatHeader component
      const givenNotifyOnLogout = jest.fn();
      const givenChatHeader = (
        <HashRouter>
          <ChatHeader
            notifyOnLogout={givenNotifyOnLogout}
            startNewConversation={givenStartNewConversation}
            notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
          />
        </HashRouter>
      );
      // AND the chat header is rendered
      render(givenChatHeader);
      // AND the user button is clicked
      const userButton = screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_BUTTON_USER);
      fireEvent.click(userButton);
      // AND the context menu is opened
      await waitFor(() => {
        expect(ContextMenu).toHaveBeenCalledWith(
          expect.objectContaining({
            anchorEl: userButton,
            open: true,
          }),
          {}
        );
      });

      // WHEN the settings menu item is clicked
      const settingsMenuItem = screen.getByTestId(MENU_ITEM_ID.SETTINGS_SELECTOR);
      fireEvent.click(settingsMenuItem);
      // AND the context menu is closed
      await waitFor(() => {
        expect(ContextMenu).toHaveBeenCalledWith(
          expect.objectContaining({
            anchorEl: null,
            open: false,
          }),
          {}
        );
      });

      // THEN expect the user to navigate to the settings page
      const navigate = useNavigate();
      expect(navigate).toHaveBeenCalledWith(routerPaths.SETTINGS);
    });

    test("should call start new conversation when the start new conversation menu item is clicked", async () => {
      // GIVEN a ChatHeader component
      const givenNotifyOnLogout = jest.fn();
      const givenStartNewConversation = jest.fn();
      const givenChatHeader = (
        <HashRouter>
          <ChatHeader
            notifyOnLogout={givenNotifyOnLogout}
            startNewConversation={givenStartNewConversation}
            notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
          />
        </HashRouter>
      );

      // AND the chat header is rendered
      render(givenChatHeader);

      // AND the user button is clicked
      const userButton = screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_BUTTON_USER);
      fireEvent.click(userButton);

      // AND the context menu is opened
      await waitFor(() => {
        expect(ContextMenu).toHaveBeenCalledWith(
          expect.objectContaining({
            anchorEl: userButton,
            open: true,
          }),
          {}
        );
      });

      // WHEN the start new conversation menu item is clicked
      const startNewConversationMenuItem = screen.getByTestId(MENU_ITEM_ID.START_NEW_CONVERSATION);
      fireEvent.click(startNewConversationMenuItem);

      // THEN expect the start new conversation function to be called
      expect(givenStartNewConversation).toHaveBeenCalled();
    });

    test("should close the context menu when notifyOnClose is called", async () => {
      // GIVEN a ChatHeader component
      const givenNotifyOnLogout = jest.fn();
      const givenChatHeader = (
        <HashRouter>
          <ChatHeader
            notifyOnLogout={givenNotifyOnLogout}
            startNewConversation={givenStartNewConversation}
            notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
          />
        </HashRouter>
      );
      // AND the chat header is rendered
      render(givenChatHeader);
      // AND the user button is clicked
      const userButton = screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_BUTTON_USER);
      fireEvent.click(userButton);

      // WHEN the notifyOnClose is called
      act(() => {
        const mock = (ContextMenu as jest.Mock).mock;
        mock.lastCall[0].notifyOnClose();
      });

      // THEN expect the auth menu to be closed
      /**
       - the component renders for the first time
       - then the menu is opened. 2nd render
       - then the menu is closed. 3rd render
       **/
      expect(ContextMenu).toHaveBeenNthCalledWith(3, expect.objectContaining({ anchorEl: null, open: false }), {});
    });

    test("should call notifyOnExperiencesDrawerOpen when the experiences menu item is clicked", async () => {
      // GIVEN a ChatHeader component
      const givenNotifyOnExperiencesDrawerOpen = jest.fn();
      const givenChatHeader = (
        <HashRouter>
          <ChatHeader
            notifyOnLogout={jest.fn()}
            startNewConversation={jest.fn()}
            notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
          />
        </HashRouter>
      );
      // AND the chat header is rendered
      render(givenChatHeader);
      // AND the context menu is opened
      const userButton = screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_BUTTON_USER);
      fireEvent.click(userButton);

      // WHEN the experiences menu item is clicked
      const experiencesMenuItem = screen.getByTestId(MENU_ITEM_ID.EXPERIENCES_BUTTON);
      fireEvent.click(experiencesMenuItem);

      // THEN expect notifyOnExperiencesDrawerOpen to be called
      expect(givenNotifyOnExperiencesDrawerOpen).toHaveBeenCalled();
    });
  });

  describe("context menu item tests", () => {
    test.each([
      ["online", true],
      ["offline", false],
    ])(
      "should render the context menu with the correct menu items when browser is %s",
      async (_description, browserIsOnline) => {
        mockBrowserIsOnLine(browserIsOnline);
        // GIVEN a ChatHeader component
        const givenNotifyOnLogout = jest.fn();
        const givenStartNewConversation = jest.fn();
        const givenNotifyOnExperiencesDrawerOpen = jest.fn();
        const givenChatHeader = (
          <HashRouter>
            <ChatHeader
              notifyOnLogout={givenNotifyOnLogout}
              startNewConversation={givenStartNewConversation}
              notifyOnExperiencesDrawerOpen={givenNotifyOnExperiencesDrawerOpen}
            />
          </HashRouter>
        );
        // AND the chat header is rendered
        render(givenChatHeader);
        // AND the user button is clicked
        const userButton = screen.getByTestId(DATA_TEST_ID.CHAT_HEADER_BUTTON_USER);
        fireEvent.click(userButton);

        // THEN expect the context menu to be visible
        await waitFor(() => {
          expect(ContextMenu).toHaveBeenCalledWith(
            expect.objectContaining({
              anchorEl: userButton,
              open: true,
              items: expect.arrayContaining([
                expect.objectContaining({
                  id: MENU_ITEM_ID.START_NEW_CONVERSATION,
                  text: "start new conversation",
                  disabled: !browserIsOnline,
                }),
                expect.objectContaining({
                  id: MENU_ITEM_ID.SETTINGS_SELECTOR,
                  text: "settings",
                  disabled: !browserIsOnline,
                }),
                expect.objectContaining({
                  id: MENU_ITEM_ID.LOGOUT_BUTTON,
                  text: "logout",
                  disabled: !browserIsOnline,
                }),
              ]),
            }),
            {}
          );
        });
        // AND the context menu to contain the correct menu items
        const contextMenu = screen.getByTestId("mock-context-menu");
        expect(contextMenu).toBeInTheDocument();
        expect(screen.getByTestId(MENU_ITEM_ID.START_NEW_CONVERSATION)).toBeInTheDocument();
        expect(screen.getByTestId(MENU_ITEM_ID.SETTINGS_SELECTOR)).toBeInTheDocument();
        expect(screen.getByTestId(MENU_ITEM_ID.LOGOUT_BUTTON)).toBeInTheDocument();
      }
    );
  });
});
