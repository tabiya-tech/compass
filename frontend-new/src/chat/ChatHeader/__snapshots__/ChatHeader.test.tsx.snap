// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader should render the Chat Header with exploredExperiencesNotification not shown 1`] = `
<div
  class="MuiBox-root css-69i1ev"
  data-testid="chat-header-container-7413b63a-887b-4f41-b930-89e9770db12b"
>
  <a
    aria-current="page"
    class="active"
    data-testid="chat-header-logo-link-7413b63a-887b-4f41-b930-89e9770db12b"
    href="#/"
    style="line-height: 0;"
  >
    <img
      alt="Compass Logo"
      data-testid="chat-header-logo-7413b63a-887b-4f41-b930-89e9770db12b"
      height="48"
      src="/compass.svg"
    />
  </a>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1m1rxeh-MuiTypography-root"
  >
    Compass
  </h1>
  <div
    class="MuiBox-root css-1n05jrb"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1672xf7-MuiButtonBase-root-MuiIconButton-root"
      data-testid="chat-header-button-experiences-7413b63a-887b-4f41-b930-89e9770db12b"
      tabindex="0"
      title="view experiences"
      type="button"
    >
      <span
        class="MuiBadge-root css-1c32n2y-MuiBadge-root"
        data-testid="animated-badge-a348462e-a393-4445-9a3b-c9603e878ce7"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
          data-testid="chat-header-icon-experiences-7413b63a-887b-4f41-b930-89e9770db12b"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 12h4v1.5h-4zm0 3h4v1.5h-4z"
          />
          <path
            d="M20 7h-5V4c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2m-9 0V4h2v5h-2zm9 13H4V9h5c0 1.1.9 2 2 2h2c1.1 0 2-.9 2-2h5z"
          />
          <circle
            cx="9"
            cy="13.5"
            r="1.5"
          />
          <path
            d="M11.08 16.18c-.64-.28-1.34-.43-2.08-.43s-1.44.15-2.08.43c-.56.24-.92.78-.92 1.39V18h6v-.43c0-.61-.36-1.15-.92-1.39"
          />
        </svg>
        <span
          class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-ohrnoc-MuiBadge-badge"
        />
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1672xf7-MuiButtonBase-root-MuiIconButton-root"
      data-testid="chat-header-button-user-7413b63a-887b-4f41-b930-89e9770db12b"
      tabindex="0"
      title="user info"
      type="button"
    >
      <img
        alt="User Icon"
        data-testid="chat-header-icon-user-7413b63a-887b-4f41-b930-89e9770db12b"
        src="/user-icon.svg"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    data-testid="b7499b01-8082-4209-8667-c7d559a70caf-menu"
  >
    <div
      data-testid="start-new-conversation-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      start new conversation
    </div>
    <div
      data-testid="settings-selector-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      settings
    </div>
    <div
      data-testid="logout-button-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      logout
    </div>
    ;
  </div>
  <div
    data-testid="anonymous-account-conversion-dialog-a4b3c2d1-e5f6-4g7h-8i9j-0k1l2m3n4o5p"
  />
</div>
`;

exports[`ChatHeader should render the Chat Header with exploredExperiencesNotification shown 1`] = `
<div
  class="MuiBox-root css-69i1ev"
  data-testid="chat-header-container-7413b63a-887b-4f41-b930-89e9770db12b"
>
  <a
    aria-current="page"
    class="active"
    data-testid="chat-header-logo-link-7413b63a-887b-4f41-b930-89e9770db12b"
    href="#/"
    style="line-height: 0;"
  >
    <img
      alt="Compass Logo"
      data-testid="chat-header-logo-7413b63a-887b-4f41-b930-89e9770db12b"
      height="48"
      src="/compass.svg"
    />
  </a>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1m1rxeh-MuiTypography-root"
  >
    Compass
  </h1>
  <div
    class="MuiBox-root css-1n05jrb"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1672xf7-MuiButtonBase-root-MuiIconButton-root"
      data-testid="chat-header-button-experiences-7413b63a-887b-4f41-b930-89e9770db12b"
      tabindex="0"
      title="view experiences"
      type="button"
    >
      <span
        class="MuiBadge-root css-1c32n2y-MuiBadge-root"
        data-testid="animated-badge-a348462e-a393-4445-9a3b-c9603e878ce7"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
          data-testid="chat-header-icon-experiences-7413b63a-887b-4f41-b930-89e9770db12b"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 12h4v1.5h-4zm0 3h4v1.5h-4z"
          />
          <path
            d="M20 7h-5V4c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2m-9 0V4h2v5h-2zm9 13H4V9h5c0 1.1.9 2 2 2h2c1.1 0 2-.9 2-2h5z"
          />
          <circle
            cx="9"
            cy="13.5"
            r="1.5"
          />
          <path
            d="M11.08 16.18c-.64-.28-1.34-.43-2.08-.43s-1.44.15-2.08.43c-.56.24-.92.78-.92 1.39V18h6v-.43c0-.61-.36-1.15-.92-1.39"
          />
        </svg>
        <span
          class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-9cqolc-MuiBadge-badge"
        >
          <span
            class="css-1tmou61"
          >
            1
          </span>
        </span>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1672xf7-MuiButtonBase-root-MuiIconButton-root"
      data-testid="chat-header-button-user-7413b63a-887b-4f41-b930-89e9770db12b"
      tabindex="0"
      title="user info"
      type="button"
    >
      <img
        alt="User Icon"
        data-testid="chat-header-icon-user-7413b63a-887b-4f41-b930-89e9770db12b"
        src="/user-icon.svg"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    data-testid="b7499b01-8082-4209-8667-c7d559a70caf-menu"
  >
    <div
      data-testid="start-new-conversation-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      start new conversation
    </div>
    <div
      data-testid="settings-selector-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      settings
    </div>
    <div
      data-testid="logout-button-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      logout
    </div>
    ;
  </div>
  <div
    data-testid="anonymous-account-conversion-dialog-a4b3c2d1-e5f6-4g7h-8i9j-0k1l2m3n4o5p"
  />
</div>
`;

exports[`ChatHeader should show feedback button when sentry is initialized 1`] = `
<div
  class="MuiBox-root css-69i1ev"
  data-testid="chat-header-container-7413b63a-887b-4f41-b930-89e9770db12b"
>
  <a
    aria-current="page"
    class="active"
    data-testid="chat-header-logo-link-7413b63a-887b-4f41-b930-89e9770db12b"
    href="#/"
    style="line-height: 0;"
  >
    <img
      alt="Compass Logo"
      data-testid="chat-header-logo-7413b63a-887b-4f41-b930-89e9770db12b"
      height="48"
      src="/compass.svg"
    />
  </a>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1m1rxeh-MuiTypography-root"
  >
    Compass
  </h1>
  <div
    class="MuiBox-root css-1n05jrb"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1672xf7-MuiButtonBase-root-MuiIconButton-root"
      data-testid="chat-header-button-experiences-7413b63a-887b-4f41-b930-89e9770db12b"
      tabindex="0"
      title="view experiences"
      type="button"
    >
      <span
        class="MuiBadge-root css-1c32n2y-MuiBadge-root"
        data-testid="animated-badge-a348462e-a393-4445-9a3b-c9603e878ce7"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
          data-testid="chat-header-icon-experiences-7413b63a-887b-4f41-b930-89e9770db12b"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 12h4v1.5h-4zm0 3h4v1.5h-4z"
          />
          <path
            d="M20 7h-5V4c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2m-9 0V4h2v5h-2zm9 13H4V9h5c0 1.1.9 2 2 2h2c1.1 0 2-.9 2-2h5z"
          />
          <circle
            cx="9"
            cy="13.5"
            r="1.5"
          />
          <path
            d="M11.08 16.18c-.64-.28-1.34-.43-2.08-.43s-1.44.15-2.08.43c-.56.24-.92.78-.92 1.39V18h6v-.43c0-.61-.36-1.15-.92-1.39"
          />
        </svg>
        <span
          class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-ohrnoc-MuiBadge-badge"
        >
          <span
            class="css-1tmou61"
          >
            0
          </span>
        </span>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1672xf7-MuiButtonBase-root-MuiIconButton-root"
      data-testid="chat-header-button-feedback-7413b63a-887b-4f41-b930-89e9770db12b"
      tabindex="0"
      title="give feedback"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
        data-testid="chat-header-icon-feedback-7413b63a-887b-4f41-b930-89e9770db12b"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2m0 14H5.17l-.59.59-.58.58V4h16zm-9-4h2v2h-2zm0-6h2v4h-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1672xf7-MuiButtonBase-root-MuiIconButton-root"
      data-testid="chat-header-button-user-7413b63a-887b-4f41-b930-89e9770db12b"
      tabindex="0"
      title="user info"
      type="button"
    >
      <img
        alt="User Icon"
        data-testid="chat-header-icon-user-7413b63a-887b-4f41-b930-89e9770db12b"
        src="/user-icon.svg"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    data-testid="b7499b01-8082-4209-8667-c7d559a70caf-menu"
  >
    <div
      data-testid="start-new-conversation-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      start new conversation
    </div>
    <div
      data-testid="settings-selector-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      settings
    </div>
    <div
      data-testid="report-bug-button-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      report a bug
    </div>
    <div
      data-testid="logout-button-7413b63a-887b-4f41-b930-89e9770db12b"
    >
      logout
    </div>
    ;
  </div>
  <div
    data-testid="anonymous-account-conversion-dialog-a4b3c2d1-e5f6-4g7h-8i9j-0k1l2m3n4o5p"
  />
</div>
`;
